From dfc27599aa7f6cdbe5c92f959f5177e083c4ce5e Mon Sep 17 00:00:00 2001
From: Eric Anholt <eric@anholt.net>
Date: Mon, 14 Dec 2009 13:12:45 -0800
Patch-mainline: 2.6.33
References: fate#310916
Git-commit: 656cb79322319a7bbafec7912d262142e9a38bc0
Subject: [PATCH 0208/2588] drm/i915: In the debugfs interface, unmap our
 address instead of the page's.

Fixes a BUG_ON in kmap_atomic for the following atomic mapping with
USER0 type.

Signed-off-by: Eric Anholt <eric@anholt.net>
(cherry picked from commit 656cb79322319a7bbafec7912d262142e9a38bc0)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_debugfs.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index af5e1e8..58fd37c 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -272,7 +272,7 @@ static void i915_dump_pages(struct seq_file *m, struct page **pages, int page_co
 		mem = kmap_atomic(pages[page], KM_USER0);
 		for (i = 0; i < PAGE_SIZE; i += 4)
 			seq_printf(m, "%08x :  %08x\n", i, mem[i / 4]);
-		kunmap_atomic(pages[page], KM_USER0);
+		kunmap_atomic(mem, KM_USER0);
 	}
 }
 
-- 
1.7.6

